### ISSUE TITLE
Wait for DOM repaint when un-disabling the infiniteScroll

### ISSUE THREAD

**Issue by magrabow** [sroze/ngInfiniteScroll#91]
Currently the `infiniteScroll` directive behaves in a way that it _watches_ the `infiniteScrollDisabled` attribute and upon change immediately fires the `handler()` function.

This is fine in general, but fails in a particular use case, that in my opinion is rather common:
- User has a property on scope that indicates that initial data is loading from the server, and uses it in a directive: `infinite-scroll-disabled="isLoading"`
- Response with data returns, and in a callback user sets `isLoading=false`
- The **$watch** on `infiniteScrollDisabled` fires the `handler()` function
- Unfortunately the DOM had no time to repaint, the page still shows with empty container element
- The just fired `handler()` function determines that bottom of the container is visible so it initiates another load of data. **-->** This behaviour is actually incorrect, as the bottom of the container is visible only because DOM hasn't been repainted yet with the newly fetched data inside that would expand the whole container. 

The proposed fix is to wrap the invocation of `handler()` in a `$timeout()` call, that would allow the DOM to repaint with the newly fetched data before actually determining the height of the container. For example like this:

```
return $timeout(function() {
    return handler();
}, 0);
```

So basically just do the same as with in the return statement of the `link` function.

This is theoretically possible to allow this DOM repaint by wrapping with `$timeout()` the statement in a controller when changing the value of `isLoading` property, but that seems hacky to perform this just for the sake of correct behaviour of the `infiniteScroll` directive, and I see no negative impact of having the directive do it on its own, even if sometimes it would be redundant.
